src/orion/serving/storage_resource.py

```diff
@@ -0,0 +1,232 @@
+
+    Properties:
+    - task_id: task ID, used to identify the task in web API
+    - notifications: stream handler with shared queue to capture task messages
+    - progress_message: latest progress message
+    - progress_value: latest progress (0 <= floating value <= 1)
+    - completed: shared status: 0 for running, -1 for failure, 1 for success
+
+    There is two ways to monitor task:
+    - either get messages collected in stream handler queue.
+      Stream handler collects all messages printed or logged in task.
+    - either regularly check latest message in progress_message.
+      Progress message only describe the latest step running in task.
+    """
+
+    # String representation of task status
+    IMPORT_STATUS = {0: "active", -1: "error", 1: "finished"}
+
+    def __init__(self):
+        self.task_id = str(uuid.uuid4())
+        self.notifications = Notifications()
+        self.progress_message = multiprocessing.Array("c", 512)
+        self.progress_value = multiprocessing.Value("d", 0.0)
+        self.completed = multiprocessing.Value("i", 0)
+
+    def set_progress(self, message: str, progress: float):
+        self.progress_message.value = message.encode()
+        self.progress_value.value = progress
+
+    def is_completed(self):
+        """Return True if task is completed"""
+        return self.completed.value
+
+    def set_completed(self, success=True):
+        """Set task terminated status"""
+        self.completed.value = 1 if success else -1
+
+
+def _import_data(task: ImportTask, storage, load_host, resolve, name, version):
+    """Function to run import task.
+
+    Set stream handler, launch load_database and set task status.
+    """
+    try:
+        print("Import starting.", task.task_id)
+        logging.basicConfig(stream=task.notifications, force=True, level=logging.INFO)
+        load_database(
+            storage,
+            load_host,
+            resolve,
+            name,
+            version,
+            progress_callback=task.set_progress,
+        )
+        task.set_completed(success=True)
+    except Exception as exc:
+        traceback.print_tb(exc.__traceback__)
+        print("Import error.", exc)
+        # Add error message to shared messages
+        task.notifications.write(f"Error: {exc}")
+        task.set_completed(success=False)
+    finally:
+        # Remove imported files
+        os.unlink(load_host)
+        lock_file = f"{load_host}.lock"
+        if os.path.exists(lock_file):
+            os.unlink(lock_file)
+        print("Import terminated.")
+
+
+class StorageResource:
+    """Handle requests for the dump/ REST endpoint"""
+
+    def __init__(self, storage):
+        self.storage = storage
+        self.current_task: ImportTask = None
+
+    def on_get_dump(self, req: Request, resp: Response):
+        """Handle the GET requests for dump/"""
+        name = req.get_param("name")
+        version = req.get_param_as_int("version")
```

Comment:
Does in handle `None`?

Reply:
Yes ! By default, it returns None if parameter is not found and not required: https://falcon.readthedocs.io/en/2.0.0/api/request_and_response.html#falcon.Request.get_param_as_int